enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/assets/OCIPropertiesProvider.java

```diff
@@ -62,16 +44,12 @@ public class OCIPropertiesProvider implements CommandProvider {
     private static final Logger LOG = Logger.getLogger(OCIPropertiesProvider.class.getName());
     private static final String  GET_DB_CONNECTION = "nbls.db.connection"; //NOI18N
 
-    private static final boolean POSIX = FileSystems.getDefault().supportedFileAttributeViews().contains("posix");  // NOI18N
-    private static final EnumSet<PosixFilePermission> readWritePosix = EnumSet.of(OWNER_READ, OWNER_WRITE);
-    private static final EnumSet<AclEntryPermission> readOnlyAcl = EnumSet.of(READ_ACL, READ_ATTRIBUTES, WRITE_ATTRIBUTES, READ_DATA, READ_NAMED_ATTRS, DELETE, SYNCHRONIZE);
-
-    // temporary directory location
-    private static final Path tmpdir = Path.of(System.getProperty("java.io.tmpdir"));       // NOI18N
+    private final ConfigFileGenerator configFileGenerator;
 
     public OCIPropertiesProvider() {
+        this.configFileGenerator = new ConfigFileGenerator("db-", ".properties", GET_DB_CONNECTION, true); // NOI18N
         try {
-            deleteOldFiles(generateDirPath());
+            configFileGenerator.deleteOldFiles();
```

Comment:
Suppose 2 NBLS servers run at the same time, e.g. two vscode windows, per-workspace NBLS (= 2 instances). `deleteOldFiles` removes files underneath `generateDirPath` which does not use random seed ... so it seems that `deleteOldFiles` in `NBLS #1` constructs the same path as `NBLS #2` is using and deletes the contents ?

Reply:
I guess you are right, one vscode instance will consume the path to the file, and the other will delete it.
I think that this deletion can be omitted since later we are deleting it anyways -> `filePath.toFile().deleteOnExit();` ?

Reply:
I'd give each NBLS process its own (random) temporary dir under java.io.tmpdir to play within. `deleteOnExit` is nice fallback, should also work on directories.